### PR TITLE
feat: add maven datasource support to semanticPrefixFixDepsChoreOthers

### LINKED ISSUE #13711: maven depType should be compliant with preset semanticPrefixFixDepsChoreOthers
### What would you like Renovate to be able to do?

Hi, 

Following the work on https://github.com/renovatebot/renovate/issues/8198.

Context : Maven provides six scopes : compile (default value if not set), provided, runtime, test, system, and import.

- compile, provided, runtime, system can be considered devDependencies
- test are test dependencies
- import is used to import another pom  that is used to provide a version for imported dependencies. So it can impact dependencies with any other scopes. Generally it is used to align devDependencies versions.

Actually Renovate do not set depType when the scope is not set in the pom.

To mimics semanticPrefixFixDepsChoreOthers preset we need to inverse the logic :

```
        {
            "matchDatasources": [
                "maven"
            ],
            "matchPackagePatterns": [
                "*"
            ],
            "semanticCommitType": "fix"
        },
        {
            "matchDatasources": [
                "maven"
            ],
            "matchDepTypes": [
                "test"
            ],
            "semanticCommitType": "chore"
        }
```

Could it be improved ? Maybe the maven extract can set the depType to compile when the value is not provided in the pom. So we can have

```
        {
            "matchDatasources": [
                "maven"
            ],
            "matchPackagePatterns": [
                "*"
            ],
            "matchDepTypes": [
                "test"
            ],
            "semanticCommitType": "chore"
        },
        {
            "matchDatasources": [
                "maven"
            ],
            "matchDepTypes": [
                "compile", "provided", "runtime", "system", "import"
            ],
            "semanticCommitType": "fix"
        }
```

Going further, I think that the semanticPrefixFixDepsChoreOthers  should work out of the box with maven dep.

### If you have any ideas on how this should be implemented, please tell us here.

Either map scope value or add/modify the preset ?

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by scrocquesel**
## Changes:

Maps all maven dependencies to a depType and augment the semanticPrefixFixDepsChoreOthers to support those depType.

Renovate will set depType as (by order of precedence):

- `parent` for parent pom
- `build` for all dependencies under a build settings element (as per https://maven.apache.org/pom.html#Build_Settings)
- scope element value or `compile` if not set for all dependencies in a dependency element
- not set for all others dependencies : only [relocation](https://maven.apache.org/pom.html#relocation) as far as I can tell from the spec.

I opt to classified `import` and `parent` dependencies as `fix`. While import and parent can only declare test/plugin dependencies, it is usually not the case, and an update to those dependencies will certainly impact the builded binaries.

Note that dependencies can be in a lot of place in the pom. depType detection mechanism may not be robust to pom schema change but the existing guard on `modelVersion` shall mitigate if it eventually change one day or another.

I haven't test the preset itself but double check the value. Maybe worth taking a look, just in case.

## Context:

Closes #13711 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/default.ts [hunk 1]
```diff
@@ -199,6 +199,25 @@ export const presets: Record<string, Preset> = {
         matchDepTypes: ['dependencies', 'require'],
         semanticCommitType: 'fix',
       },
+      {
+        matchDatasources: ['maven'],
+        matchPackagePatterns: ['*'],
```
Comment (rarkins) [inline lib/config/presets/internal/default.ts:204]:
```suggestion
```

File: lib/config/presets/internal/default.ts [hunk 2]
```diff
@@ -199,6 +199,25 @@ export const presets: Record<string, Preset> = {
...
+        matchDepTypes: ['test', 'build'],
+        semanticCommitType: 'chore',
+      },
+      {
+        matchDatasources: ['maven'],
+        matchPackagePatterns: ['*'],
```
Comment (rarkins) [inline lib/config/presets/internal/default.ts:210]:
```suggestion
```

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — DISMISSED** [review]

File: lib/config/presets/internal/default.ts [hunk 3]
```diff
@@ -199,6 +199,23 @@ export const presets: Record<string, Preset> = {
         matchDepTypes: ['dependencies', 'require'],
         semanticCommitType: 'fix',
       },
+      {
+        matchDatasources: ['maven'],
+        matchDepTypes: ['test', 'build'],
+        semanticCommitType: 'chore',
```
Comment (viceice) [inline lib/config/presets/internal/default.ts:205]: Isn't `chore` the default, so we can remove this rule?

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

**Comment by viceice** [comment]
Needs test fixes

File: lib/config/presets/internal/default.ts (hunk 3 above)
Comment (scrocquesel) [reply lib/config/presets/internal/default.ts:205]: Yes, it is purely to be explicit

Comment (viceice) [reply lib/config/presets/internal/default.ts:205]: Remove, as it's default in previous rule

**Comment by scrocquesel** [comment]
@rarkins may I ask for a quick review again. thanks

**Comment by viceice** [comment]
> @rarkins may I ask for a quick review again. thanks

👉 https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#re-requesting-a-review 🤗

**Review by viceice — APPROVED** [review]

**Pull request merged**